docs/guides/dropout.rst

```diff
@@ -0,0 +1,147 @@
+Applying dropout
+================
+
+This guide provides an overview of how to apply
+`dropout <https://jmlr.org/papers/volume15/srivastava14a/srivastava14a.pdf>`__
+using :meth:`flax.linen.Dropout`
+
+Dropout is a stochastic regularization technique that randomly removes hidden
+and visible units in a network.
+
+Throughout the guide, you will be able to compare code examples with and without
+Flax ``Dropout``.
+
+.. testsetup::
+
+  import flax.linen as nn
+  import jax.numpy as jnp
+  import jax
+  import optax
+
+Split the PRNG
+**************
+
+Dropout is a random operation, requiring a PRNG state. Flax (like JAX) uses
+`Threefry <https://github.com/google/jax/blob/main/docs/jep/263-prng.md>`__ PRNG
+that is splittable.
+
+**Note:** Recall that JAX has an explicit way of giving you PRNG keys:
+you can fork the main PRNG state (such as ``key = jax.random.PRNGKey(seed=0)``)
+into multiple new PRNG keys with ``key, subkey = jax.random.split(key)``. You
+can refresh your memory in
+`🔪 JAX - The Sharp Bits 🔪 Randomness and PRNG keys <https:/t/jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#random-numbers>`__.
+
+You split the PRNG into keys, including one for dropout, as follows::
+
+  seed = 0
+  root_key = jax.random.PRNGKey(seed=seed)
+  main_key, params_key, dropout_key = jax.random.split(key=root_key, num=3)
+
+Each of the Flax PRNG streams has a name, such as the ``'params'`` stream for
+initializing parameters, and ``'dropout'`` for dropout.
```

Comment:
Good idea.

With some minor refactoring of your suggestion:

```
In Flax, you provide PRNG streams with names, so that you can use them later in
your ``Module``. For example, you pass the stream ``'params'`` for
initializing parameters, and ``'dropout'`` for applying :meth:`flax.linen.Dropout`.
```